### ISSUE TITLE
[no-type-alias] Unhandled in union types are not allowed using "readonly string[]"

### ISSUE THREAD

**Issue by nicolashenry** [typescript-eslint/typescript-eslint#1981]
**Repro**

```JSON
{
    "rules": {
        "@typescript-eslint/no-type-alias": [
            "error",
            {
                "allowAliases": "in-unions-and-intersections"
            }
        ]
    }
}
```

```TS
export type MyType = string | readonly string[];
```

**Expected Result**
No Error

**Actual Result**
`error  Unhandled in union types are not allowed  @typescript-eslint/no-type-alias`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.31.0` |
| `@typescript-eslint/parser`        | `2.31.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.2` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
Small oversight - should be trivial to add support for this. Happy to accept a PR!
